src/features/RenameSymbol.ts

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import vscode = require("vscode");
+import { RequestType } from "vscode-languageclient";
+import { LanguageClientConsumer } from "../languageClientConsumer";
+import { RenameProvider, WorkspaceEdit, TextDocument, CancellationToken, Position,Uri,Range } from "vscode";
+import type { LanguageClient } from "vscode-languageclient/node";
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface IRenameSymbolRequestArguments {
+    FileName?:string
+    Line?:number
+    Column?:number
+    RenameTo?:string
+}
+interface IPrepareRenameSymbolRequestArguments {
+    FileName?:string
+    Line?:number
+    Column?:number
+    RenameTo?:string
+}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface TextChange {
+
+    newText: string;
+    startLine: number;
+    startColumn: number;
+    endLine: number;
+    endColumn: number;
+}
+interface ModifiedFileResponse{
+    fileName: string;
+    changes : TextChange[]
+}
+
+interface IRenameSymbolRequestResponse {
+    changes : ModifiedFileResponse[]
+}
+
+interface IPrepareRenameSymbolRequestResponse {
+    message : string
+}
+
+export const RenameSymbolRequestType = new RequestType<IRenameSymbolRequestArguments, IRenameSymbolRequestResponse, void>("powerShell/renameSymbol");
+export const PrepareRenameSymbolRequestType = new RequestType<IPrepareRenameSymbolRequestArguments, IPrepareRenameSymbolRequestResponse, void>("powerShell/PrepareRenameSymbol");
+
+export class RenameSymbolFeature extends LanguageClientConsumer implements RenameProvider {
+    public override onLanguageClientSet(_languageClient: LanguageClient): void {
+        throw new Error("Method not implemented.");
```

Comment:
Was there something that needs to be registered here? Otherwise this is just throwing all the time because I'm pretty sure this gets called on startup.

Reply:
Fixed, thanks!